File: cpp_solutions/1208.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int n = s.size();
        vector<int> diff(n, 0);
        for (int i = 0; i < n; ++i) {
            diff[i] = abs(s[i] - t[i]);
        }

        int curr = 0;
        int left = 0;
        int res = 0;
        for (int right = 0; right < n; ++right) {
            curr += diff[right];
            while (curr > maxCost) {
                curr -= diff[left];
                left++;
            }
            res = max(res, right - left + 1);
        }
        return res;
    }
};

File: rust_solutions/1208.rs
impl Solution {
    pub fn equal_substring(s: String, t: String, max_cost: i32) -> i32 {
        let n = s.len();
        let mut diff = vec![0; n];

        let sbytes = s.as_bytes();
        let tbytes = t.as_bytes();

        for i in 0..n {
            diff[i] = (u8::abs_diff(sbytes[i], tbytes[i])) as usize;
        }

        let mut curr: i32 = 0;
        let mut left: usize = 0;
        let mut res: i32 = 0;
        for right in 0..n {
            curr += diff[right] as i32;
            while curr > max_cost {
                curr -= diff[left] as i32;
                left += 1;
            }
            res = i32::max(res, (right - left + 1) as i32);
        }
        return res;
    }
}
